compiler/il/OMRBlock.cpp

```diff
@@ -1472,11 +1472,35 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
             // If we can not find a register, store it into temp slot.
             if (!iter->second.second)
                {
-               iter->second.second = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectLoad(value->getDataType()), 0, ref);
+               TR::DataType valDataType = value->getDataType();
+               TR::Node *convertedValueNode  = value;
+               // In case for a given node, if the load/store (Minimum 4 bytes for alignment) size in memory is not same as node's data type,
```

Comment:
the 4 byte minimum is an OpenJ9 thing - such a comment does not belong in OMR. The key point here is that if the data type of the stack load/store does not match the node data type then we need to do the correct conversion.

Reply:
Forgot to fixup comment yesterday, changed comment suitable to OMR in https://github.com/eclipse/omr/pull/4551/commits/76c0d39b5d8f0cea95c5e33dc06c87c55a7d4816